react/react_test.go

```diff
@@ -21,47 +24,96 @@ func TestTestVersion(t *testing.T) {
 	}
 }
 
+func assertCellValue(t *testing.T, c Cell, expected int, explanation string) {
+	observed := c.Value()
+	if observed != expected {
+		t.Fatalf("%s: expected %d, got %d", explanation, expected, observed)
+	}
+}
+
 // Setting the value of an input cell changes the observable Value()
 func TestSetInput(t *testing.T) {
 	r := New()
 	i := r.CreateInput(1)
-	if i.Value() != 1 {
-		t.Fatalf("i.Value() doesn't match initial value")
-	}
+	assertCellValue(t, i, 1, "i.Value() doesn't match initial value")
 	i.SetValue(2)
-	if i.Value() != 2 {
-		t.Fatalf("i.Value() doesn't match changed value")
-	}
+	assertCellValue(t, i, 2, "i.Value() doesn't match changed value")
 }
 
-// The value of a compute cell is determined by the value of the dependencies.
-func TestBasicCompute(t *testing.T) {
+// The value of a compute 1 cell is determined by the value of the dependencies.
+func TestBasicCompute1(t *testing.T) {
 	r := New()
 	i := r.CreateInput(1)
 	c := r.CreateCompute1(i, func(v int) int { return v + 1 })
-	if c.Value() != 2 {
-		t.Fatalf("c.Value() isn't properly computed based on initial input cell value")
-	}
+	assertCellValue(t, c, 2, "c.Value() isn't properly computed based on initial input cell value")
 	i.SetValue(2)
-	if c.Value() != 3 {
-		t.Fatalf("c.Value() isn't properly computed based on changed input cell value")
-	}
+	assertCellValue(t, c, 3, "c.Value() isn't properly computed based on changed input cell value")
+}
+
+// The value of a compute 2 cell is determined by the value of the dependencies.
+func TestBasicCompute2(t *testing.T) {
+	r := New()
+	i1 := r.CreateInput(1)
+	i2 := r.CreateInput(2)
+	c := r.CreateCompute2(i1, i2, func(v1, v2 int) int { return v1 | v2 })
+	assertCellValue(t, c, 3, "c.Value() isn't properly computed based on initial input cell values")
+	i1.SetValue(4)
+	assertCellValue(t, c, 6, "c.Value() isn't properly computed when first input cell value changes")
+	i2.SetValue(8)
+	assertCellValue(t, c, 12, "c.Value() isn't properly computed when second input cell value changes")
 }
 
-// Compute cells can depend on other compute cells.
+// Compute 2 cells can depend on compute 1 cells.
 func TestTopology(t *testing.T) {
```

Comment:
Perhaps a name change is in order now, since there are now multiple topologies as opposed to before when this was the only topology test. Perhaps `TestCompute2Diamond` because the topology is kind of in a diamond shape. Suggestions very welcome. Or if you think it's fine I'll leave it, but I suspect I should change.